python/lsst/ip/isr/binExposureTask.py

```diff
@@ -0,0 +1,202 @@
+        instance. This enables both to be dynamically set at runtime,
+        allowing BinExposureTask to work with different types of
+        Exposures such as postISRCCD, calexp, deepCoadd_calexp, etc.
+
+        Parameters
+        ----------
+        config : `BinExposureConfig`
+            A config for `BinExposureTask`.
+        """
+        super().__init__(config=config)
+        if config and config.exposureDimensions != self.inputExposure.dimensions:
+            self.dimensions.clear()
+            self.dimensions.update(config.exposureDimensions)
+            self.inputExposure = cT.Input(
+                name=self.inputExposure.name,
+                doc=self.inputExposure.doc,
+                storageClass=self.inputExposure.storageClass,
+                dimensions=frozenset(config.exposureDimensions),
+            )
+            self.binnedExposure = cT.Output(
+                name=self.binnedExposure.name,
+                doc=self.binnedExposure.doc,
+                storageClass=self.binnedExposure.storageClass,
+                dimensions=frozenset(config.exposureDimensions),
+            )
+        if config and config.exposureStorageClass != self.inputExposure.storageClass:
+            self.inputExposure = cT.Input(
+                name=self.inputExposure.name,
+                doc=self.inputExposure.doc,
+                storageClass=config.exposureStorageClass,
+                dimensions=self.inputExposure.dimensions,
+            )
+            self.binnedExposure = cT.Output(
+                name=self.binnedExposure.name,
+                doc=self.binnedExposure.doc,
+                storageClass=config.exposureStorageClass,
+                dimensions=self.binnedExposure.dimensions,
+            )
+
+
+class BinExposureConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=BinExposureConnections
+):
+    """Config for BinExposureTask"""
+    exposureDimensions = pexConfig.ListField(
+        # Sort to ensure default order is consistent between runs
+        default=sorted(BinExposureConnections.dimensions),
+        dtype=str,
+        doc="Override for the dimensions of the input and binned exposures.",
+    )
+    exposureStorageClass = pexConfig.Field(
+        default='ExposureF',
+        dtype=str,
+        doc="Override the storageClass of the input and binned exposures."
+    )
+    binFactor = pexConfig.Field(
+        dtype=int,
+        doc="Binning factor applied to both spatial dimensions.",
+        default=8,
+        check=lambda x: x > 1,
+    )
+
+
+class BinExposureTask(pipeBase.PipelineTask):
+    """Perform an nxn binning of an Exposure dataset type.
+
+    The binning factor is the same in both spatial dimensions (i.e.,
+    an nxn binning is performed). Each of the input Exposure's image
+    arrays are binned by the same factor.
+    """
+    ConfigClass = BinExposureConfig
+    _DefaultName = "binExposure"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
```

Comment:
This looks like it's just doing the standard `runQuantum` work, so I think you can drop this (and it'll use the parent class version).